pdp/policy.go

```diff
@@ -243,6 +250,42 @@ func (p *Policy) delChild(ID string) (*Policy, error) {
 	return p.updatedCopy(rules, algorithm), nil
 }
 
+// DepthMarshal implements StorageMarshal
+func (p Policy) DepthMarshal(out io.Writer, depth int) error {
+	if depth < 0 {
+		return fmt.Errorf("depth must be >= 0, got %d", depth)
+	}
+	if p.hidden {
+		return errHiddenPolicy
+	}
+	_, err := out.Write([]byte(fmt.Sprintf(policyFmt, p.ord, p.id)))
```

Comment:
Please double check that `Write` isn't allowed to write less than passed without any error. If it's allowed you need to make sure that all bytes have been written before going further.

Also `p.id` can be any string and contain JSON special characters as `"`, `,` and so on. So it's better to use json.Marshal here to format fields as `p.ord` and `p.id` and than put it them with `{"ord": %s, "id": %s, "rules": [` format.

Reply:
I see that [io.Writer description](https://golang.org/pkg/io/#Writer) says: "Write must return a non-nil error if it returns n < len(p)". So we can rely just on error returned from Write.